alacritty/src/window.rs

```diff
@@ -361,6 +366,27 @@ impl Window {
         self.window().wayland_display()
     }
 
+    #[cfg(not(any(target_os = "macos", target_os = "windows")))]
+    pub fn set_wayland_theme(&mut self, config: &Config) {
+        let hovered_close_icon = config.colors.normal().red;
+        let hovered_maximize_icon = config.colors.normal().green;
+        let hovered_minimize_icon = config.colors.normal().yellow;
+        let foreground = config.colors.primary.foreground;
+        let background = config.colors.primary.background;
+        let dim_foreground =
+            config.colors.primary.dim_foreground.unwrap_or_else(|| foreground * 0.66);
+
+        let theme = AlacrittyWaylandTheme {
+            foreground,
+            background,
+            dim_foreground,
+            hovered_close_icon,
+            hovered_minimize_icon,
+            hovered_maximize_icon,
+        };
+        self.window().set_wayland_theme(theme);
+    }
```

Comment:
@kchibisov I thought you said this wouldn't need to be handled inside of Alacritty. It seems unfortunate to need to make all these choices for our users. What if my applications should all use blue to maximize?

Reply:
You don't have control over it anyways, since it's a client side decorations. Before we had random white Qt'ish color scheme, now we have something that matches content. You can't just draw GTK decorations, etc. Providing direct config option isn't an option, so we should derive it our of users colorscheme.

Reply:
I feel like I may be misremembering a conversation... but I was under the impression we were against CSDs, in favor of SSDs. I can maybe understand wanting an escape hatch from poorly configured SSDs, but I really wish it wasn't inside out codebase.

Here's what I'm remembering: https://github.com/alacritty/alacritty/pull/2047#issuecomment-461925986

Reply:
We support both CSD and SSD this option is for CSD theming, since we control them. We do support CSD and we're not against them. So on sway you'll use SSD on GNOME CSD.

Reply:
Sorry if this is obvious, but where do we "support both CSD and SSD" ATM?

Reply:
FWIW, my main issue here is with the plan for configuration of things like this. I don't really want to block enhancements that make Alacritty look better for users, since people will be happier to use Alacritty if it looks good.

Reply:
>Sorry if this is obvious, but where do we "support both CSD and SSD" at the moment?

SCTK handles them for us.

Reply:
OK, point taken. I'd like to propose moving these out into their own `wayland.rs` file somewhere however.

Reply:
I'd just keep it simple with a `wayland_theme.rs` file for now and if we ever need more platform specific modules, we can separate it more.